test/heavymock/exporter_impl.go

```diff
@@ -32,13 +33,23 @@ func NewRecordExporter(importerServer *ImporterServer) *RecordExporter {
 func (r *RecordExporter) Export(request *exporter.GetRecords, stream exporter.RecordExporter_ExportServer) error {
 	r.mux.Lock()
 	defer r.mux.Unlock()
-	records := r.importerServer.GetUnsentRecords()
+	records := []*exporter.Record{}
+	for _, r := range r.importerServer.GetUnsentRecords() {
+		if r.Record.ID.Pulse() != request.PulseNumber {
+			continue
+		}
+		records = append(records, r)
+	}
+
+	r.importerServer.MarkAsSent(records)
+
+	records = append(records, testutils.GenerateRecordInNextPulse(request.PulseNumber)) // marker of pulse's end
```

Comment:
I don't agree this should be here. The test itself should control which pulses were sent.
Adding this nextPulse, the test gets unexpected behaviour meaning that it doesn't know last pulse number sent